test/edu/csus/ecs/pc2/core/ParseArgumentsTest.java

```diff
@@ -72,31 +92,66 @@ public void testIsOptPresent() {
         assertNull(opt2+" option should be null value", parseArguments.getOptValue(opt2));
     }
     
+    public void testAreOptsValid() {
+
+        String opt = "--login";
+        String opt2 = "--debad";
+        String opt3 = "--nogui";
+        String optp = "-pass";
+        String value = "name";
+        String opt4 = "--bill";
+
+        String[] required = { opt, optp };
+        String[] args = { opt, value, opt3, opt4, optp };
+        String[] allowedOpts = { opt, opt3, optp };
+        ParseArguments parseArguments = null;
+        
+        try {
+            parseArguments = new ParseArguments(args, required);
+        } catch (IllegalArgumentException exc) {
+            fail("Un-expected IllegalArgumentException for ParseArguments(args, required)");
+        }
+//        parseArguments.dumpArgs(System.out);
+
+        parseArguments.setAllowedOpts(allowedOpts);
+        assertTrue(opt + " option should be allowed", parseArguments.isAllowedOption(opt));
+        assertFalse(opt2 + " option should not be allowed", parseArguments.isAllowedOption(opt2));
+        assertTrue(optp + " option should be allowed", parseArguments.isAllowedOption(optp));
+        assertFalse(value + " option should not be allowed ", parseArguments.isAllowedOption(value));
+        
+        assertFalse(opt4 + " option should not be allowed", parseArguments.isAllowedOption(opt4));
+        assertTrue(opt3 + " option should be allowed", parseArguments.isAllowedOption(opt3));
+    }
+    
     /**
      * main test routine for ParseArgs <br>
      *
      * @param args
      *            java.lang.String[]
      */
     public static void main(String[] args) {
-        ParseArguments pa = new ParseArguments();
-        pa.loadArgs(args);
-        pa.dumpArgs(System.out);
-        System.out.println();
-        
-        String [] reqArgs = { "--l", "--file" };
-
-        System.out.print("Using required arguments:");
-        for (String s : reqArgs){
-            System.out.print(" "+s);
-        }
-        System.out.println();
-        
-        pa = new ParseArguments();
-        pa.setRequireArgOpts(reqArgs);
-        pa.loadArgs(args);
-        pa.dumpArgs(System.out);
-
+        ParseArgumentsTest t = new ParseArgumentsTest();
+        t.testIsOptPresent();
+        t.testAreOptsValid();
+//      if(false) {
+//      ParseArguments pa = new ParseArguments();
+//      pa.loadArgs(args);
+//      pa.dumpArgs(System.out);
+//      System.out.println();
+//      
+//      String [] reqArgs = { "--l", "--file" };
+//
+//      System.out.print("Using required arguments:");
+//      for (String s : reqArgs){
+//          System.out.print(" "+s);
+//      }
+//      System.out.println();
+//      
+//      pa = new ParseArguments();
+//      pa.setRequireArgOpts(reqArgs);
+//      pa.loadArgs(args);
+//      pa.dumpArgs(System.out);
+//      }
```

Comment:
Personally I would rather see blocks of deprecated code like the removed, rather than just "commented out"; the presence of the old code makes it harder to read the current version.  Especially when the old code is characterized by things like blocks that start with `if (false)...`.   

However, this doesn't rise to the level of needing to deny approval of the PR...

Reply:
Agreed, old/unused code should be removed.   My comment/problem that 1 - if(false) dead code
2 the dead code would be found if grep'ing the code which requires extra analysis/time to
to identify that it is old/dead code

Reply:
Removed vestigial code and pushed.